common/menu-list.c

```diff
@@ -150,3 +150,53 @@ int menuScan(const char* target) {
     return 0;
 }
 
+
+int themeMenuScan(const char* target) {
+    menuClear();
+    if (chdir(target) < 0) return 1;
+    if (getcwd(s_menu[!s_curMenu].dirname, PATH_MAX+1) == NULL)
+        return 1;
+    DIR* dir;
+    struct dirent* dp;
+    char tmp_path[PATH_MAX+1];
+    dir = opendir(s_menu[!s_curMenu].dirname);
+    if (!dir) return 2;
+
+    while ((dp = readdir(dir)))
+    {
+        menuEntry_s* me = NULL;
+
+        bool shortcut = false;
+        if (dp->d_name[0]=='.')
+            continue;
+
+        memset(tmp_path, 0, sizeof(tmp_path));
+        snprintf(tmp_path, sizeof(tmp_path)-1, "%s/%s", s_menu[!s_curMenu].dirname, dp->d_name);
+
+        const char* ext = getExtension(dp->d_name);
+        char* name = removeExtension(dp->d_name);
+        replaceCharacter(name,'_',' ');
+        if(strcmp(dp->d_name,"theme.cfg")==0)//This theme is already the currently applied theme, dont load it
+            continue;
+        if (strcasecmp(ext, ".cfg")==0)
+            me = menuCreateEntry(ENTRY_TYPE_THEME);
+
+        if (!me)
+            continue;
+
+        strncpy(me->path, tmp_path, sizeof(me->path)-1);
+        me->path[sizeof(me->path)-1] = 0;
+        if (menuEntryLoad(me,(const char*)name, shortcut))
+            menuAddEntry(me);
+        else
+            menuDeleteEntry(me);
+        free((void*)name);//This was allocated by calloc and copied over by strcpy
```

Comment:
No need to cast to void*. All (non-const) pointers are implicitly convertible to void*.

Reply:
i do this to get rid of the warnings, i know it doesnt matter, but i will get rid of it if it bothers you.